### PR TITLE
Add new channel_types field to CommandOption

### PR DESCRIPTION
**Pull request by davfsa**

### CODE REVIEW

File: hikari/commands.py [hunk 1]
```diff
@@ -136,6 +137,11 @@ class CommandOption:
     options: typing.Optional[typing.Sequence[CommandOption]] = attr.field(default=None, repr=False)
     """Sequence of up to (and including) 25 of the options for this command option."""
 
+    channel_types: typing.Optional[typing.Sequence[typing.Union[channels.ChannelType, int]]] = attr.field(
+        default=None, repr=False
+    )
+    """If a channel type option, the valid channel types that this option will accept."""
```
Comment (FasterSpeeding) [inline hikari/commands.py:143]: `valid channel types` you shouldn't be expecting to get invalid channel types here so thats redundant

Comment (davfsa) [reply hikari/commands.py:143]:
```suggestion
    """If a channel type option, the channel types that this option will accept."""
```

File: hikari/commands.py [hunk 2]
```diff
@@ -136,6 +137,11 @@ class CommandOption:
     options: typing.Optional[typing.Sequence[CommandOption]] = attr.field(default=None, repr=False)
     """Sequence of up to (and including) 25 of the options for this command option."""
 
+    channel_types: typing.Optional[typing.Sequence[typing.Union[channels.ChannelType, int]]] = attr.field(
+        default=None, repr=False
+    )
+    """If a channel type option, the channel types that this option will accept."""
```
Comment (FasterSpeeding) [inline hikari/commands.py:143]: its also worth noting that if this is None for a channel type option then it accepts all option types

Comment (davfsa) [reply hikari/commands.py:143]:
```suggestion
    """The channel types that this option will accept.
    
    If `builtins.None`, then all channel types will be accepted.
    """
```

**Review by FasterSpeeding — DISMISSED** [review]

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**